### ISSUE TITLE
Cannot specify data connectors which do not conform to DataConnectorConfigSchema

### ISSUE THREAD

**Issue by sfrdmn** [fivetran/great_expectations#2546]
**Description**

If I want to use a data connector with an API not conforming explicitly to `DataConnectorConfigSchema` (eg even the packaged `ConfiguredAssetSqlDataConnector`), and configure it at the level of the `DataContext` (eg `great_expectations.yml`), non-conforming properties are not provided to the connector constructor

**To Reproduce**

```python
from great_expectations import DataContext

DataContext().test_yaml_config("""
class_name: ConfiguredAssetSqlDataConnector
module_name: great_expectations.datasource.data_connector
datasource_name: good_source_of_data
name: good_connector
data_assets:
  orders:
    table_name: orders
    schema_name: online_business
""")
```

Observe that the connector as well as its data assets are instantiated

```python
from great_expectations import DataContext

DataContext().test_yaml_config("""
class_name: Datasource
name: good_source_of_data
module_name: great_expectations.datasource
execution_engine:
  class_name: SqlAlchemyExecutionEngine
  module_name: great_expectations.execution_engine
  connection_string: postgresql://google.com
data_connectors:
  good_data_connector:
    class_name: ConfiguredAssetSqlDataConnector
    module_name: great_expectations.datasource.data_connector
    data_assets:
      orders:
        table_name: orders
        schema_name: online_business
""")
```

Observe that when instantiated _through_ a data context and data source, the data connector is instantiated without any data assets (and to note: the `data_assets` property does not conform to `DataConnectorConfigSchema`)

**Expected behavior**

I can successfully configure a `ConfiguredAssetSqlDataConnector` with data assets via `DataContext`

**Environment**

MacOS
python 3.7.3
great_expectations 0.13.12

**Additional Context**

The specific problem seems to be a call to `DataContextConfigSchema.dump` when instantiating the datasource. As far as I understand, `dump` will not yield unknown keys, even if the schema itself is configured to allow unknown keys. At least, that's the behavior I seem to be seeing. There are quite a few `dump` calls peppered in `DataContext`, so not sure exactly what to think here

**Comment by eugmandel** [comment]
@sfrdmn This is not the intended behavior and indeed seems to be a bug. Thank you for reporting this issue. We are adding it to the core team's queue.

**Comment by eugmandel** [comment]
@sfrdmn Sorry I did not notice this earlier - could you please try using "assets" key instead of "data_assets" in your configuration? Please report back if this fixes the issue for you. Thank you!

**Comment by sfrdmn** [comment]
@eugmandel Nope, doesn't work:

```
Couldn't instantiate class : ConfiguredAssetSqlDataConnector with config :
yadda yadda
__init__() got an unexpected keyword argument 'assets'
```

**Comment by sfrdmn** [comment]
Oh and to note: I modified my local copy of GE to add schema definitions in conformance with the API exposed by  `ConfiguredAssetSqlDataConnector` (eg data connector schema allowing `{"data_assets": { "name": { "schema_name" }}}`) and _that_ works

**Comment by eugmandel** [comment]
Thank you for trying this, @sfrdmn ! All right - this is definitely a bug. Labeled "core-team-priority".

**Comment by sfrdmn** [comment]
The specific issue I ran into here appears to be addressed by #2665 (This issue is expressed as a schema problem only really because I didn't know what the intended behavior was)

**Comment by talagluck** [comment]
Thanks! Closing, as this is resolved by #2665.
